frontend_tests/casper_tests/05-subscriptions.js

```diff
@@ -131,25 +131,43 @@ casper.then(function () {
 });
 casper.then(function () {
     casper.click('#create_stream_button');
-    casper.fill('form#stream_creation_form', {stream_name: '  '});
-    casper.click('form#stream_creation_form button.btn.btn-primary');
+    casper.waitUntilVisible("form#stream_creation_form", function () {
+        casper.fill('form#stream_creation_form', {stream_name: '  '});
+        casper.click('form#stream_creation_form button.btn.btn-primary');
+    });
 });
 casper.then(function () {
     casper.waitForSelectorText('#stream_name_error', 'A stream needs to have a name', function () {
         casper.test.assertTextExists('A stream needs to have a name', "Can't create a stream with an empty name");
-        casper.click('form#stream_creation_form button.btn.btn-default');
-        casper.fill('form#add_new_subscription', {stream_name: '  '});
-        casper.click('#create_stream_button');
-        casper.fill('form#stream_creation_form', {stream_name: 'Waseemio'});
-        casper.click('form#stream_creation_form button.btn.btn-primary');
+        casper.waitUntilVisible("form#stream_creation_form", function () {
+            casper.click('form#stream_creation_form button.btn.btn-default');
+            casper.fill('form#add_new_subscription', {stream_name: '  '});
+            casper.click('#create_stream_button');
+        });
+        casper.waitUntilVisible("form#stream_creation_form", function () {
```

Comment:
This wait will probably be not needed because now the form is already visible, so just add this:
```
casper.click('form#stream_creation_form button.btn.btn-default'); 
casper.fill('form#stream_creation_form', {stream_name: 'Waseemio@'}); 
casper.click('form#stream_creation_form button.btn.btn-primary');
```

Reply:
Or perhaps we do. I think `casper.click('form#stream_creation_form button.btn.btn-default');` will close the form? Can you verify the behavior? If `btn-default` closes the form then we will need a wait.

Reply:
`casper.click('form#stream_creation_form button.btn.btn-default');` does close the form. So, it would need I wait. Thanks! :)